pyblish_ftrack/ftrack_event_plugin_path/publish.py

```diff
@@ -141,18 +141,31 @@ def _discoverApplications(self, path=''):
         if path:
             launchArguments = [path]
 
-        return self._searchFilesystem(
+        icon = 'https://raw.githubusercontent.com/pyblish/pyblish-ftrack/'
+        icon += 'master/pyblish_ftrack/ftrack_event_plugin_path/icon.png'
+
+        application = self._searchFilesystem(
             expression=['C:\\', 'Python*', 'python.exe'],
             label='Publish',
             applicationIdentifier='publish',
-            icon="https://raw.githubusercontent.com/pyblish/pyblish-ftrack/master/pyblish_ftrack/ftrack_event_plugin_path/icon.png",
+            icon=icon,
             launchArguments=launchArguments
         )
 
+        for path in os.environ['PATH'].split(os.pathsep):
+            if 'python' in path.lower():
+                path = os.path.join(path, 'python.exe')
```

Comment:
For other OSs you would just query which OS you are in, and implement whatever logic there is to find the executable.